src/apps/HeyGPT/HeyGPT/ViewModels/LetsChatViewModel.cs

```diff
@@ -195,5 +280,133 @@ private void AddErrorMessageToHistory(string errorMessage)
         });
     }
 
+    private async Task SendMessageInternal(string userMessage, ChatCharacterDetails? desiredCharacterDetails = null)
+    {
+        AddUserMessageToHistory(userMessage);
+
+        try
+        {
+            var characterDetails = desiredCharacterDetails ?? _characterService.GetSelectedCharacterDetails();
+
+            var extraContext = await GetClipboardContext().ConfigureAwait(true);
+
+            ShouldIncludeClipboardContext = false; // Reset the extra context option after it is used so the user is forced to do it each time.
+
+            var response = await _openAIService.SendPrompt(characterDetails, userMessage, extraContext.Text).ConfigureAwait(true);
+
+            var message = $"{response.Content} [{response.MetaData.Usage.TotalTokens}]";
+
+            AddRobotMessageToHistory(response.CharacterType, message);
+        }
+        catch (Exception e)
+        {
+            AddErrorMessageToHistory(e.ToString());
+        }
+    }
+
+    public static (string Character, string Message) ParseHeyProtocolUri(Uri uri)
+    {
+        // Ensure the URI is in the correct format
+        if (uri.Scheme != "hey" || !uri.Host.Equals("send", StringComparison.OrdinalIgnoreCase))
+        {
+            throw new ArgumentException("The URI is not in the correct format.");
+        }
+
+        // Get the segments of the URI
+        var segments = uri.Segments;
+
+        if (segments.Length < 3)
+        {
+            throw new ArgumentException("The URI does not contain enough segments.");
+        }
+
+        // Extract the Character and Base64 encoded message from the segments
+        var character = segments[1].Trim('/');
+        var base64Message = segments[2];
+
+        // Decode the Base64 message
+        var message = Encoding.UTF8.GetString(Convert.FromBase64String(base64Message));
+
+        return (character, message);
+    }
+
+    private static readonly Random Random = new();
+
+    public static int GetRandomIndex(int lowerBound, int upperBound)
```

Comment:
The file related methods are all just a simple way to manage structured vs. unstructured and our desire to just prove out the file activation concept.   This needs to be cleaned up, perhaps into a UX service.